### ISSUE TITLE
False Positive on Vaadin versions from 10 onwards

### ISSUE THREAD

**Issue by TatuLund** [dependency-check/DependencyCheck#2191]
The OWASP tool is a bit faulty with new Vaadin versions . It matches all dependencies starting with "vaadin-" (or maybe "vaadin") having version number less than 7 with our old Vaadin 6 reported security issues. Hence it is showing large number of false positives as all our components bundled in Vaadin 10+ start with "vaadin-" and have version number starting with either 1 or 2.

**Comment by aikebah** [comment]
Hi @TatuLund , I took a quick look and to me it looks like a hard-to-resolve issue. How would you propose that dependency-check  links the version of those components to the corresponding Vaadin release? As NVD only has vaadin:vaadin as part of it's CPE list a CVE in a vaadin component will likely be linked to the corresponding Vaadin release.

**Comment by TatuLund** [comment]
You can see the problem by looking our BOM https://mvnrepository.com/artifact/com.vaadin/vaadin-bom/14.0.5

Nowadays our framework is modular. User will include it in project by adding dependency to vaadin-bom, the latest version is 14.0.5. The BOM includes lot of sub-modules with different versions as you see.

**Comment by jeremylong** [comment]
The real issue for dependency-check is that the components are not versioned at the same time as the version of Vaadin.  I have added a few suppression rules that should resolve most of these issues.  We may need a few more - but I added these:

```xml
    <suppress base="true">
        <notes><![CDATA[
        FP per #2191
        ]]></notes>
        <packageUrl regex="true">^pkg:maven/com\.vaadin\.(addon|external).*$</packageUrl>
        <cpe>cpe:/a:vaadin:vaadin</cpe>
    </suppress>
    <suppress base="true">
        <notes><![CDATA[
        FP per #2191
        ]]></notes>
        <packageUrl regex="true">^pkg:maven/com\.vaadin/.*(flow|theme|mixin|style|widget).*$</packageUrl>
        <cpe>cpe:/a:vaadin:vaadin</cpe>
    </suppress>
    <suppress base="true">
        <notes><![CDATA[
        FP per #2191
        ]]></notes>
        <packageUrl regex="true">^pkg:maven/org\.webjars\.(bower|bowergithub|npm).*$</packageUrl>
        <cpe>cpe:/a:vaadin:vaadin</cpe>
    </suppress>
```

**Comment by TatuLund** [comment]
Thank you for fast response on this, it is very much appreciated. I tested version 5.2.2 with one of my test projects and report is not showing those false positives anymore.
